source/use_cases/aws-restaurant-management-demo/lib/manager-stack.ts

```diff
@@ -0,0 +1,223 @@
+/**
+ *  Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+// Imports
+import * as cdk from '@aws-cdk/core';
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as dynamodb from '@aws-cdk/aws-dynamodb';
+import * as apigateway from '@aws-cdk/aws-apigateway';
+import * as events from '@aws-cdk/aws-events';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as sftasks from '@aws-cdk/aws-stepfunctions-tasks';
+import * as s3 from '@aws-cdk/aws-s3';
+import { CognitoToApiGatewayToLambda } from '@aws-solutions-constructs/aws-cognito-apigateway-lambda';
+import { LambdaToDynamoDB } from '@aws-solutions-constructs/aws-lambda-dynamodb';
+import { LambdaToS3 } from '@aws-solutions-constructs/aws-lambda-s3';
+import { LambdaToSns } from '@aws-solutions-constructs/aws-lambda-sns';
+import { EventsRuleToLambda } from '@aws-solutions-constructs/aws-events-rule-lambda';
+import { LambdaToStepFunction } from '@aws-solutions-constructs/aws-lambda-step-function';
+
+// Properties for the manager-stack
+export interface Resources {
```

Comment:
Why define this as its own interface and pass as a 4th argument to the construct? The conventional approach would be to create a dedicated Props type for this construct that includes these values, wouldn't it?